### ISSUE TITLE
Port already allocated if requesting port dynamically after requesting specific static port

### ISSUE THREAD

**Issue by tiborvass** [docker/docker#7158]
on master, b2d/aufs

With a fresh daemon:
```
$ docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS                   NAMES
$ docker run -d -p 49153:80 busybox:latest top
5371b43c052299c01911e5611294cd4d3b2fb7dbf61363c5feb23d84f5476e74
$ docker run -d -p 80 busybox:latest top
c2bb694236aab82b0a50422a9da7488c4538d62c4afe52d64ee324c7ceb246c2
2014/07/22 16:10:01 Error response from daemon: Cannot start container c2bb694236aab82b0a50422a9da7488c4538d62c4afe52d64ee324c7ceb246c2: port is already mapped to ip
$ docker run -d -p 80 busybox:latest top
d2e95896ad9a675e7ba9495fb881ed9c7178420b91e541a626e3f08ad1d5d408
```

Works on a second run.

**Comment by bru7us** [comment]
Giving this a bump (diverted to this open issue from #8723). We are about to deploy some stuff into production and are using an F5 BIG-IP as a proxy for all of our micro-services. On deployment, we let Docker allocate a dynamic port, then capture this data as input to our BIG-IP pool configuration.

If we have to restart a container or host, having the port change will cause our pool member to go "offline", and to recover we would be forced to manually create a new pool member with the newly allocated port.

In short, I'd like to vote for having a restarted container keep the same port allocation that it was initially given. The alternative is for us to write code that selects and manually allocated a specific port to a container at "run" time - which seems pointless given that Docker already provides that functionality with a simple 'EXPOSE' instruction.

**Comment by gdm85** [comment]
@bru7us I see #8723 is marked as dup of this one - although I think it's a separate issue, and rather the logic that leads to #8723 has been introduced as a workaround to the issue here.

What I do [here](https://github.com/gdm85/docker-fw/blob/master/state.go#L80) for example is restore the host config (port bindings) across restarts of the containers.

**Comment by tiborvass** [comment]
I'm closing this issue, since I could not reproduce anymore.

@bru7us what you are looking for is #10052. I am against the flag, but in favor of allocating port on create.

**Comment by bru7us** [comment]
Ah-hah!! Thanks Tibor!! I'll keep a close eye on that PR then.
